### PR TITLE
implement RomanNumerals

### PR DESCRIPTION
**Pull request by SteffenKonrad**
solved issue  #23

### CODE REVIEW

**Comment by nicolaiparlog** [comment]
Green check mark, nice. :smiley: Your Roman Numerals solution uses a couple of advanced features (lambdas, multiple returns), which makes me wonder whether you wrote it entirely yourself. If yes, that's great, if not, that's also not a problem. Finding a solution and applying it to your concrete code base is almost as good. Just try to make sure to understand as many details as possible.

File: src/main/kotlin/org/codefx/demo/bingen/energy_model/math/RomanNumerals.kt [hunk 1]
```diff
@@ -1,5 +1,32 @@
 package org.codefx.demo.bingen.energy_model.math
 
-fun asRoman(n: Int): String {
-    return ""
+val decimalToRomanMapping : Map<Int, String> = mapOf(
+
```
Comment (nicolaiparlog) [inline src/main/kotlin/org/codefx/demo/bingen/energy_model/math/RomanNumerals.kt]: It's a matter of taste but I would not have empty lines like these.

File: src/main/kotlin/org/codefx/demo/bingen/energy_model/math/RomanNumerals.kt [hunk 2]
```diff
@@ -1,5 +1,32 @@
...
+        Pair(4, "IV"),
+        Pair(1, "I")
+)
+
+
+fun asRoman(n: Int): String {          //the function
+
+    var decimalRemainder = n           // Declaration of the variable
+    var romanOutput = ""               // "" for the correct output
+
```
Comment (nicolaiparlog) [inline src/main/kotlin/org/codefx/demo/bingen/energy_model/math/RomanNumerals.kt]: Two empty lines in a row are particularly uncommon.

**Comment by SteffenKonrad** [comment]
Servus,

Knowledge from EDV-Erweiterungen (WP-Fach) & Internet ^^. I still have the blank lines and a few comments added.

I have started a second pull request, with a simple built version.

cheers Steffen

**Comment by SteffenKonrad** [comment]
Is it possible to delete commit 4 + 5?!?!

**Comment by nicolaiparlog** [comment]
Merged the other one, so we don't have overly advanced features in the code base. Good work. :+1:

**Pull request closed without merging**